### ISSUE TITLE
Custom EXE name crashing

### ISSUE THREAD

**Issue by RuthlessJailer** [Ryochan7/DS4Windows#2549]
**Describe the bug**
When editing the exe's name, the program will crash upon starting.
**To Reproduce**
1. Open DS4Windows **not as administrator**
2. Go to the settings tab
3. Disable "run at startup"
4. Set "to run as program"
5. Set "Custom Exe Name"
6. __Tick "Minimize to Taskbar"__
7. Stop/start, then quit DS4Windows
8. Attempt to open the new EXE
9. It crashes

**Expected behavior**
It does not crash

**Screenshots and Logs**
```txt
2022-08-31 21:25:47.5026|INFO|DS4Windows version 3.1.3
2022-08-31 21:25:47.5026|INFO|DS4Windows exe file: DS4Win.exe
2022-08-31 21:25:47.5026|INFO|DS4Windows Assembly Architecture: x64
2022-08-31 21:25:47.5026|INFO|OS Version: Microsoft Windows NT 10.0.19044.0
2022-08-31 21:25:47.5026|INFO|OS Product Name: Windows 10 Home
2022-08-31 21:25:47.5026|INFO|OS Release ID: 2009
2022-08-31 21:25:47.5026|INFO|System Architecture: x64
2022-08-31 21:25:47.5026|INFO|Logger created
2022-08-31 21:25:48.3376|INFO|Running as User
2022-08-31 21:25:49.3600|INFO|Starting...
2022-08-31 21:25:49.3600|INFO|Using output KB+M handler: FakerInput 0.1.0.0
2022-08-31 21:25:49.3600|INFO|Connection to ViGEmBus 1.17.333.0 established
2022-08-31 21:25:49.3600|INFO|Searching for controllers...
2022-08-31 21:25:49.3600|INFO|Using Shared Mode
2022-08-31 21:25:49.4923|INFO|Found Controller: 48:18:8D:E3:ED:C4 (BT) (DualSense)
2022-08-31 21:25:49.5692|INFO|Plugging in virtual DS4 Controller
2022-08-31 21:25:49.5692|INFO|Associate DS4 Controller in slot #1 for input DualSense controller #1
2022-08-31 21:25:49.5934|INFO|Controller 1 is using Profile "Default". Battery level is 37%
2022-08-31 21:25:49.6315|ERROR|Thread Crashed with message TrayIcon is not created.
2022-08-31 21:25:49.6472|ERROR|System.InvalidOperationException: TrayIcon is not created.
   at H.NotifyIcon.Core.TrayIcon.EnsureCreated() in /_/src/libs/H.NotifyIcon/Core/TrayIcon.cs:line 609
   at H.NotifyIcon.Core.TrayIcon.ShowNotification(String title, String message, NotificationIcon icon, Nullable`1 customIcon, Boolean largeIcon, Boolean sound, Boolean respectQuietTime, Boolean realtime, Nullable`1 timeout) in /_/src/libs/H.NotifyIcon/Core/TrayIcon.cs:line 491
   at H.NotifyIcon.TaskbarIcon.ShowNotification(String title, String message, NotificationIcon icon, Icon customIcon, Boolean largeIcon, Boolean sound, Boolean respectQuietTime, Boolean realtime, Nullable`1 timeout) in /_/src/libs/H.NotifyIcon.Shared/TaskbarIcon.Notifications.cs:line 71
   at DS4WinWPF.DS4Forms.MainWindow.<>c__DisplayClass30_0.<ShowNotification>b__0()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
2022-08-31 21:25:49.6472|ERROR|Thread Crashed with message TrayIcon is not created.
2022-08-31 21:25:49.6472|ERROR|System.InvalidOperationException: TrayIcon is not created.
   at H.NotifyIcon.Core.TrayIcon.EnsureCreated() in /_/src/libs/H.NotifyIcon/Core/TrayIcon.cs:line 609
   at H.NotifyIcon.Core.TrayIcon.ShowNotification(String title, String message, NotificationIcon icon, Nullable`1 customIcon, Boolean largeIcon, Boolean sound, Boolean respectQuietTime, Boolean realtime, Nullable`1 timeout) in /_/src/libs/H.NotifyIcon/Core/TrayIcon.cs:line 491
   at H.NotifyIcon.TaskbarIcon.ShowNotification(String title, String message, NotificationIcon icon, Icon customIcon, Boolean largeIcon, Boolean sound, Boolean respectQuietTime, Boolean realtime, Nullable`1 timeout) in /_/src/libs/H.NotifyIcon.Shared/TaskbarIcon.Notifications.cs:line 71
   at DS4WinWPF.DS4Forms.MainWindow.<>c__DisplayClass30_0.<ShowNotification>b__0()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run()
   at DS4WinWPF.App.Main()
2022-08-31 21:25:49.6472|INFO|Stopping Virtual Output Controllers
2022-08-31 21:25:49.6472|INFO|Closing connection to ViGEmBus
2022-08-31 21:25:49.6472|WARN|48:18:8D:E3:ED:C4 disconnected due to read failure: 997
2022-08-31 21:25:49.6587|INFO|Disassociate DS4 Controller from slot #1 for input DualSense controller #1
2022-08-31 21:25:49.6587|INFO|Unplugging virtual DS4 Controller
2022-08-31 21:25:49.6587|INFO|Controller 1 was removed or lost connection
2022-08-31 21:25:49.7142|INFO|Stopping DS4 Input Controllers
2022-08-31 21:25:49.7142|INFO|Stopped DS4Windows

```

**Desktop (please complete the following information):**
 - Controller Make and Model: Sony DualSense (CFI-ZCT1W)
 - OS: Windows 10 Home
 - DS4Windows Version: 3.1.3

**Additional context**
__It appears to have to do with the taskbar icon__

**Comment by Yohoki** [comment]
[[Notice] Read this before posting a new Discussion or Issue ticket requesting help (Troubleshooting tips)](https://github.com/Ryochan7/DS4Windows/issues/1196)


This issue already has several reports.

https://github.com/Ryochan7/DS4Windows/issues/2545#issuecomment-1229561262 :
>There's some issues with the trayicon and custom names right now. Some of these issues have been addressed already and will be pushed in the next update. The push to the newer .Net 6 broke a couple things.
>
>Check version 3.1.4+ when they release and see if that version has fixed the issue. For now, [version 3.0.18](https://github.com/Ryochan7/DS4Windows/releases/tag/v3.0.18) is the last version before the .Net 6 changes. It _should_ still work fine for you.

**Comment by RuthlessJailer** [comment]
Sorry! I had just downloaded it and assumed latest version!

**Comment by Yohoki** [comment]
It was the latest version until about 3 hours ago. [Version 3.1.4](https://github.com/Ryochan7/DS4Windows/releases/tag/v3.1.4) was released today and has the newest fixes in it, relating to the taskbar icon and custom names. Give that a go and see if it fixes your problem. Come back and give a reply so we know if it's all good. :)

**Comment by Ryochan7** [comment]
This issue is fixed in version 3.1.4. Just need to move a portion of the TaskbarIcon workaround earlier in the code.
